stellargraph/mapper/mini_batch_node_generators.py

```diff
@@ -113,15 +114,13 @@ def __init__(self, G, clusters=1, q=1, lam=0.1, name=None):
                 )
             )
 
-        # Check if the graph has features
-        G.check_graph_for_ml()
-
         self.node_list = list(G.nodes())
 
-        # Check that there is only a single node type
-        _ = G.unique_node_type(
-            "G: expected a graph with a single node type, found a graph with node types: %(found)s"
-        )
+        # If the graph is in the StellaGraph format check that there
+        # is only a single node type and that the graph has features
+        if isinstance(G, StellarGraph):
```

Comment:
I wonder if we should do this special casing within StellarGraph and Neo4jStellarGraph. One approach might be:

- add a defaulted parameter `expensive_checks=False` to `check_graph_for_ml`
- `StellarGraph` does its normal checks for `expensive_checks=False` (and currently there's nothing extra when `expensive_checks=True`)
- `Neo4jStellarGraph` does nothing, or some O(1) work for `expensive_checks=False`, and O(graph size) work for `expensive_checks=True`

I'm not entirely sure of the best way to handle `unique_node_type` with `expensive_checks=False`, maybe something using `db.labels()` (https://neo4j.com/docs/operations-manual/current/reference/procedures/)?